### DISCUSSION TITLE
USACE Requirements for OpenDCS in CWBI:  Office-Specific Instances of OpenDCS

### DISCUSSION THREAD

**Discussion by jbatterman** [opendcs/opendcs#334] [General]
The Missouri River Basin Water Management Office has brainstormed some requirements for what is needed for USACE to operate OpenDCS in the Cloud.  We are looking for feedback on these requirements, and additions from other offices who have different data acquisition processes.  There will be a new Discussion for each requirement.

Need to have office-specific instances of OpenDCS, or the ability to filter by office ID from a national dataset.  This includes DECODES, computations, groups, algorithms, routing specifications, and all current facets of the OpenDCS toolkit.  This requirement also needs to be able to allow one DCP to be decoded by multiple offices (i.e., one station may be connected to multiple configs, platforms, etc. from different offices).  As data comes into the LRGS, the message queue needs to be read by all offices that want to collect it.  That data needs to be tagged with an office ID so that if it gets transformed differently by multiple offices it will be seen by the correct groups/computations/etc.

@MikeNeilson @hoffmanjs @opendcs/opendcs-core-devs @opendcs/opendcs-governance @opendcs/opendcs-triage

**Comment by MikeNeilson** [comment]
The CWMS implementation of OpenDCS already includes office information in minimum required tables to enable this behavior.

(Minimum here is based on relation database table normalization. It may make sense to include the information in other tables should filtering prove difficult)

That said i would likely be good for OpenDCS as a whole to incorporate some form of "organization" (for USACE that would be the office) for all implementations.


For the multiple offices using one gauge. I would argue that's actually a bit of a tricky one. 99% of the time it would likely make sense to have the same baseline math operations for that station. (like stage->flow and inst flow -> avg flow.)

I had a thought of using "CWMS" as the office idea for those but Ruth pointed out some challenges with that. It's possible that a system of "inheritance" within the organization system would make sense there.

For USACE/CWBI the primary intent would be to distinguish bulk automation (like GOES processing, or USGS/USBR/etc) from local district operations like SPK's CDEC data that's SPK/SPL only and little if any overlap.
